### ISSUE TITLE
Errors should display useful informations.

### ISSUE THREAD

**Issue by gpessa** [rsaenen/ngx-custom-validators#12]
**IMPORTANT:**
Validator like "greater than" (equalTo has the same issue) should return back the value which is suppose to be bigger of so we can include the information in the error we have to display.

**Steps to reproduce and a minimal demo**
Add a validator to the field and check the value of the error.
```
<input type="number" [(ngModel)]="model.field" name="field" #field="ngModel" [gt]="10">
<p *ngIf="field.errors?.gt">error message</p>
```

**Current behavior**
The error return a simple boolean.

**Expected/desired behavior**
```
{
  greaterThan: 3
}
```

in this way i can do something like:

```
<input type="number" [(ngModel)]="model.field" name="field" #field="ngModel" [gt]="10">
<p *ngIf="field.errors?.gt">field should be greater than {{ field.error.gt }}</p>
```

**Comment by rsaenen** [comment]
Okay, it will be released soon

**Comment by gpessa** [comment]
Love u! 
Would you like to have a list of validator that could require a bit of this refactoring?

**Comment by rsaenen** [comment]
This take a bit more time as I have to rewrite a lot a unit test, sorry

**Comment by rsaenen** [comment]
Thanks! Released with 6.0.3.

The paramater of each validator (if it has) can be accessible in the template with reason.
```html
<input type="number" [(ngModel)]="model.field" name="field" #field="ngModel" [gt]="10">
<!--> Will display : error message and must be greater than 10 <-->
<p *ngIf="field.errors?.gt">error message and must be greater than {{ field.errors?.reason }}</p>1
```

**Comment by gpessa** [comment]
Dow :(
Is different from the way the standard angular validators work.
Your way override the reasons if more than one validator is there.
This is how it looks:

Actual situation:

`
controls.errors : {
  required: true,
  gt: true,
  reason: 10
}
`

Desired situation:

`
controls.errors : {
  required: true,
  gt: {
    reason: 10
  }
}
`

in this way you can have multiple validators which do not override each-others reasons

**Comment by rsaenen** [comment]
Following the [Material Guideline](https://material.io/design/components/text-fields.html#anatomy), you shouldn't display more than one error message.

In your example, gt won't trigger an error when the field has already a required error. 

Nevertheless, put the reason in gt is (technically) a better choice.
Can you make a PR?

**Comment by rsaenen** [comment]
Released with 6.10
